### ISSUE TITLE
r/aws_lb_target_group: Check at plan-time for TCP protocol health check values

### ISSUE THREAD

**Issue by jen20** [terraform-providers/terraform-provider-aws#2783]
The following error is returned from AWS if the healthy and unhealthy threshold for a TCP check are difference:

```
* aws_lb_target_group.redacted: Error creating LB Target Group: InvalidConfigurationRequest: Health check healthy threshold and unhealthy threshold must be the same for target groups with the TCP protocol
	status code: 400, request id: 79b2a32e-eb54-11e7-9d3c-61a602e0a672
```

This would be straightforward to check at plan time in order to fail faster.

**Comment by loivis** [comment]
@jen20 Do you know any other resource that is using this kind of check during plan? `ValidateFunc` only validates the argument itself, but not to compare its value with other attribute. I don't get it from type `*schema.Schema`.

**Comment by bflad** [comment]
The plan-time validation has been merged into master and will release with v1.12.0 of the AWS provider, likely in about a week. Cheers!

**Comment by bflad** [comment]
This has been released in [version 1.12.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1120-march-23-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
